### PR TITLE
✨ Add new show more button example

### PR DESCRIPTION
**Pull request by lluerich**
Besides just fixing the broken show more button example, I added a new example that shows the built-in functionality to display a show more button via amp-list attribute.
Both examples now use the photo-stream API.

### CODE REVIEW

File: examples/source/interactivity-dynamic-content/Show_More_Button.html [hunk 1]
```diff
@@ -34,20 +34,65 @@
...
+    </div>
+    <amp-list-load-more load-more-failed>
+      ERROR
+    </amp-list-load-more>
+    <amp-list-load-more load-more-end>
+      END
+    </amp-list-load-more>
+  </amp-list>
+
   <!-- You can implement a show more button by using `amp-list` and `amp-bind`, where
```
Comment (sebastianbenz) [inline examples/source/interactivity-dynamic-content/Show_More_Button.html]: Can you add a new heading: `## Show more using amp-list and amp-bind`

Comment (sebastianbenz) [reply examples/source/interactivity-dynamic-content/Show_More_Button.html]: Can you also add a intro sentence along the lines of: "You can also manually implement the show more functionality using amp-bind and amp-state."

Comment (lluerich) [reply examples/source/interactivity-dynamic-content/Show_More_Button.html]: Would it suffice to edit the paragraph starting line 85 to something like so: "You can also manually implement a show more button by using `amp-list` and `amp-bind`, where `amp-list`[...]"

Comment (sebastianbenz) [reply examples/source/interactivity-dynamic-content/Show_More_Button.html]: Yes, + the heading

Comment (lluerich) [reply examples/source/interactivity-dynamic-content/Show_More_Button.html]: PTAL

File: examples/source/interactivity-dynamic-content/Show_More_Button.html [hunk 2]
```diff
@@ -82,7 +82,8 @@
     </amp-list-load-more>
   </amp-list>
 
-  <!-- You can implement a show more button by using `amp-list` and `amp-bind`, where
+  <!-- ## Show more using amp-list and amp-bind -->
+  <!-- You can also implement a show more button manually by using `amp-list` and `amp-bind`, where
   `amp-list` `src` data are bind to the value of an `amp-state` element. Here we are using two `amp-state`: `productsState`
```
Comment (sebastianbenz) [inline examples/source/interactivity-dynamic-content/Show_More_Button.html]: one nit (didn't see it earlier): `productsState` doesn't really make sense anymore. Use `photos` instead.

Comment (lluerich) [reply examples/source/interactivity-dynamic-content/Show_More_Button.html]: PTAL

**Pull request merged**